macros/combine_property_data.sql

```diff
@@ -3,36 +3,44 @@
 {%- endmacro -%}
 
 {% macro default__combine_property_data() %}
-
-    create schema if not exists `{{target.project}}.{{var('combined_dataset')}}`;
-
-    {# If incremental, then use static_incremental_days variable to find earliest shard to copy #}
     {% if not should_full_refresh() %}
-        {% set earliest_shard_to_retrieve = (modules.datetime.date.today() - modules.datetime.timedelta(days=var('static_incremental_days')))|string|replace("-", "")|int %}
+        {# If incremental, then use static_incremental_days variable to find earliest shard to copy #}
+        {%- set earliest_shard_to_retrieve = (modules.datetime.date.today() - modules.datetime.timedelta(days=var('static_incremental_days')))|string|replace("-", "")|int -%}
     {% else %}
-    {# Otherwise use 'start_date' variable #}
-
-        {% set earliest_shard_to_retrieve = var('start_date')|int %}
+        {# Otherwise use 'start_date' variable #}
+        {%- set earliest_shard_to_retrieve = var('start_date')|int -%}
     {% endif %}
 
     {% for property_id in var('property_ids') %}
         {%- set schema_name = "analytics_" + property_id|string -%}
+
+        {%- set combine_specified_property_data_query -%}
+            create schema if not exists `{{target.project}}.{{var('combined_dataset')}}`;
+
             {# Copy intraday tables #}
             {%- set relations = dbt_utils.get_relations_by_pattern(schema_pattern=schema_name, table_pattern='events_intraday_%', database=var('source_project')) -%}
             {% for relation in relations %}
                 {%- set relation_suffix = relation.identifier|replace('events_intraday_', '') -%}
                 {%- if relation_suffix|int >= earliest_shard_to_retrieve|int -%}
-                    CREATE OR REPLACE TABLE `{{target.project}}.{{var('combined_dataset')}}.events_intraday_{{relation_suffix}}{{property_id}}` CLONE `{{var('source_project')}}.analytics_{{property_id}}.events_intraday_{{relation_suffix}}`;
+                    create or replace table `{{target.project}}.{{var('combined_dataset')}}.events_intraday_{{relation_suffix}}{{property_id}}` clone `{{var('source_project')}}.analytics_{{property_id}}.events_intraday_{{relation_suffix}}`;
```

Comment:
It seems best to minimize the time between query generation and execution.

With the changes made in this pull request, the likelihood of this happening should decrease significantly, so it seems prudent to monitor the situation for now.

The following error occurred:

```bash
23:07:31  BigQuery adapter: https://console.cloud.google.com/bigquery?project={{target.project}}&j=bq:asia-northeast1:xxxxxxxxx-xxxx-xxxx-xxx-xxxxxxxxxxxx&page=queryresults
23:07:31  1 of 999 ERROR creating sql incremental model {{dataset_id}}.base_ga4__events ... [ERROR in 260.79s]
```

```bash
23:08:48  Completed with 1 error and 0 warnings:
23:08:48  
23:08:48  Database Error in model base_ga4__events (models/staging/base/base_ga4__events.sql)
23:08:48    Not found: Table {{target.project}}:analytics_{{property_id}}.events_intraday_20240327 was not found in location asia-northeast1 at [77:5]
```

It appears that there was a transition from `events_intraday_20240327` to `events_20240327` before referencing `{{target.project}}:analytics_{{property_id}}.events_intraday_20240327`, resulting in a `Not found` error.

Reply:
@dgitis

I've been experiencing the error mentioned above quite frequently lately, and it's been causing some trouble.
If possible, I would like to proceed with addressing this error here, but is merging this pull request difficult?
If there are any specific fixes needed, please let me know so I can take care of them.

Also, if there are any other good solutions available, I'd like to consider them.